### PR TITLE
Fix azure_rm_storageaccount failure due to new version of Azure Pytho…

### PR DESCRIPTION
**Pull request by ZhijunZhao**
…n SDK

##### SUMMARY

The failure is as blow

```
\"/tmp/ansible_frHOyW/ansible_module_azure_rm_storageaccount.py\", line 416, in create_account\n    poller = self.storage_client.storage_accounts.create(self.resource_group, self.name, parameters)\n  File \"/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/azure/mgmt/storage/v2017_06_01/operations/storage_accounts_operations.py\", line 159, in create\n    body_content = self._serialize.body(parameters, 'StorageAccountCreateParameters')\n  File \"/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/msrest/serialization.py\", line 460, in body\n    SerializationError, \"Unable to build a model: \"+str(err), err)\n  File \"/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/msrest/exceptions.py\", line 48, in raise_with_traceback\n    raise error\nmsrest.exceptions.SerializationError: Unable to build a model: <SkuName.standard_lrs: 'Standard_LRS'> is not valid value for enum <enum 'SkuName'>, DeserializationError: <SkuName.standard_lrs: 'Standard_LRS'> is not valid value for enum <enum 'SkuName'>\n"
```

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

azure_rm_storageaccount

##### ANSIBLE VERSION

```
ansible 2.5.0
  config file = None
  configured module search path = [u'/Users/zhijzhao/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.14 (default, Sep 25 2017, 09:53:22) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.37)]

```

##### ADDITIONAL INFORMATION

```

```
@zikalino @yuwzho @yaweiw

### CODE REVIEW

**Comment by yuwzho** [comment]
shipit

**Comment by yuwzho** [comment]
shipit

**Comment by ZhijunZhao** [comment]
@nitzmahone

**Comment by ZhijunZhao** [comment]
@lmazuel Please help to review the code. If you think this is the appropriate solution, I'll make similar changes to the models of other clients.

**Comment by lmazuel** [comment]
@ZhijunZhao looks good to me :)

FYI, 2017-10-01 introduced in 1.5.0 is the introduction of Storage V2:
https://docs.microsoft.com/azure/storage/common/storage-account-options#general-purpose-v2

Just so you know if someone asks for this support.

Note that this is not compatible with AzureStack, since Azure Stack has a specific set of ApiVersion supported, and 2017-10-01 is not one of them. I'm not sure how Azure Stack compat with Ansible is close or not on your roadmap, do not hesitate to talk to me when necessary. My email is \<my github alias\>@microsoft.com.

Thanks!!

File: lib/ansible/module_utils/azure_rm_common.py [hunk 1]
```diff
@@ -678,24 +676,24 @@ def create_default_securitygroup(self, resource_group, location, name, os_type,
             self.check_provisioning_state(group)
             return group
 
-        parameters = NetworkSecurityGroup()
+        parameters = self.network_models.NetworkSecurityGroup()
```
Comment (yuwzho) [inline lib/ansible/module_utils/azure_rm_common.py]: Why we fix the version in the code? That means we should use all the azure SDK model in this way not only in the azure_rm_common

**Comment by nitzmahone** [comment]
merge conflict corrected out-of-band, direct committed to devel, cherry picked to stable-2.4 via https://github.com/ansible/ansible/pull/34682 for 2.4.3rc1

**Pull request closed without merging**